control/init.sh

```diff
@@ -35,8 +34,15 @@ echo "Downloading dependencies from galaxy"
 export ANSIBLE_ROLES_PATH=~/.ansible/roles
 ansible-galaxy install -r ~/commcarehq-ansible/ansible/requirements.yml
 
+pip install -r ~/commcarehq-ansible/ansible/requirements.txt
+pip install -e ~/commcarehq-ansible/commcare-cloud
+
 # convenience: . init-ansible
 [ ! -f ~/init-ansible ] && ln -s ~/commcarehq-ansible/control/init.sh ~/init-ansible
+[ ! -d ~/.commcare-cloud ] && mkdir ~/.commcare-cloud
+[ ! -d ~/.commcare-cloud/ansible ] && ln -s ~/commcarehq-ansible/ansible ~/.commcare-cloud/
+[ ! -d ~/.commcare-cloud/vars ] && ln -s ~/commcarehq-ansible/ansible/vars ~/.commcare-cloud/
+[ ! -d ~/.commcare-cloud/inventory ] && ln -s ~/commcare-hq-deploy/fab/inventory ~/.commcare-cloud/
```

Comment:
Why is this indirection (symlinks in `~/.commcare-cloud/`) necessary? Would it work to remove them and replace `~/.commcare-cloud/` with `~/commcarehq-ansible/ansible/`?

Reply:
So, I'm not sure it's "necessary", but I decided to do it because I think it gives you something nice. In terms of why it's nice to have `~/.commcare-cloud/` at all, it makes configuring `commcare-cloud` to run on a different environment a little easier, so for example if I run it on my own machine (as I was doing) I can symlink the correct locations there and it works because `~/commcarehq-ansible/ansible/vars` for example isn't hard-coded. But maybe that wasn't your question; maybe you were asking more specifically about 3 symlinks versus one.

`ansible` and `vars` are symlinks to `~/commcarehq-ansible/ansible` and a subdirectory with the same name (respectively), but `inventory` is symlinked to a different location entirely. Could I have forgone `vars` entirely and referenced it in my code as `~/.commcare-cloud/ansible/vars`? Well, yes I could have in this case. However, `vars` and `inventory` stand out to me in that they are the only parts of the code that deal with _our_ cloud environment, as opposed to _a_ commcare cloud environment. You could imagine a world for example where those two directories live together in a private repository as opposed to living apart in two different public repos that are made up of otherwise generic code.

Not sure if that gets at what you were asking, happy to talk more!

Reply:
This seemed weird to me las tnight, but  I didn't have a different solution. What about using environment variables?

Reply:
I'd prefer to have these be options that can be passed to `commcare-cloud`. Something like `--ansible-dir`, `--ansible-vars-dir`, and `--inventory-dir` with values set here as defaults. It could use environment variables following the pattern of [some of our nose plugins](https://github.com/dimagi/commcare-hq/blob/64fabc2799cf6a2f7c49e2db8929c77eb4f86d3b/corehq/tests/noseplugins/dividedwerun.py#L35).

Having this setup/configuration in the init script means there is more scattered code to read to understand what the tool is actually doing. It feels OK for this tool to be opinionated about some of these things (at least by default) since it is specifically made to work with the commcare ansible setup—it even has "commcare" in its name.